pkg/codegen/dotnet/gen_program_test.go

```diff
@@ -1,11 +1,97 @@
 package dotnet
 
 import (
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
 	"testing"
 
+	"github.com/stretchr/testify/assert"
+
 	"github.com/pulumi/pulumi/pkg/v3/codegen/internal/test"
+	"github.com/pulumi/pulumi/pkg/v3/testing/integration"
+	"github.com/pulumi/pulumi/sdk/v3/go/common/util/executable"
 )
 
 func TestGenerateProgram(t *testing.T) {
-	test.TestProgramCodegen(t, "dotnet", GenerateProgram)
+	test.TestProgramCodegen(t,
+		test.ProgramLangConfig{
+			Language:   "dotnet",
+			Extension:  "cs",
+			OutputFile: "MyStack.cs",
+			Check:      checkDotnet,
+			GenProgram: GenerateProgram,
+		},
+	)
+}
+
+func checkDotnet(t *testing.T, path string) {
+	var err error
+	dir := filepath.Dir(path)
+
+	ex, err := executable.FindExecutable("dotnet")
+	assert.NoError(t, err, "Failed to find dotnet executable")
+
+	projectFile := filepath.Join(dir, filepath.Base(dir)+".csproj")
+	if _, err := ioutil.ReadFile(projectFile); os.IsNotExist(err) {
+		defer func() {
+			err = os.Remove(projectFile)
```

Comment:
Philosophically I have a slight preference for keeping this files in rather than always automatically deleting them, in case something is failing I would like to have the ability to interact with the generated program as a normal program, edit it and experiment with it.

Reply:
+1

Reply:
The system now creates metadata files before each test. This way after a test you can `cd` into the test area and manually build and modify the test. 

Because these files are recreated after each test, metadata files are still ephemeral.